site/src/Librecores/ProjectRepoBundle/Util/GitHubUtil.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Librecores\ProjectRepoBundle\Util;
+
+/**
+ * Helper utility service, which provides methods for operating with GitHub projects.
+ */
+
+class GitHubUtil
+{
+
+    private $githubUrl;
+    private $shieldsUrl;
+    private $shieldsExt;
+    private $sourceRepoUrl;
+    private $issueTrackerUrl;
+
+    public function __construct($githubUrl, $shieldsUrl, $shieldsExt)
+    {
+        $this->githubUrl       = $githubUrl;
+        $this->shieldsUrl      = $shieldsUrl;
+        $this->shieldsExt      = $shieldsExt;
+        $this->sourceRepoUrl   = null;
+        $this->issueTrackerUrl = null;
```

Comment:
> I kept the original separation of sourceRepo and issueTracker URLs because I liked the original intention of allowing a project to have the source repo and issue tracker separate.

Yes, this is must-have. Many big projects use external bugtrackers like JIRA

> If it starts getting complicated we could wrap it in a factory service but I don't think there are so many Github URLs to justify this. Even adding a couple more can be done quickly as it is

OK. What we may really need:
* Releases support (with proper badges, etc.)
* Maybe "Maintainers" link pointing to a GitHub team